frontend/lib/Views/AdminPage/members/add_user_modal.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:adpv_frontend/DataModels/member_summary.dart';
+import 'package:adpv_frontend/Views/AdminPage/utils.dart';
+import 'package:flutter/material.dart';
+
+import '../../../Repository/AdminRepository/admin_gateway.dart';
+import '../../../Widgets/common_widgets.dart';
+
+MaterialStateProperty<EdgeInsetsGeometry?> buttonPadding =
+    MaterialStateProperty.all(const EdgeInsets.all(20));
+
+void showAddUserModal(
+  BuildContext context,
+  AdminGateway gateway,
+  int groupId,
+  Function(int id) onProceedFunction,
+) {
+  showDialog(
+    context: context,
+    builder: (_) {
+      int selected = -1;
```

Comment:
Z tego co rozumiem, to mamy dialog z opcją wpisania adresu email z podpowiadaniem. Obecnie mamy takie założenie, że po wciśnięciu jakiejś podpowiedzi wpisuje się ona nam na input, wciskamy `Add` i dodajemy użytkownika. A co w przypadku, kiedy ktoś wpisze `email` z palca? W tym momencie wyślemy zapytanie z `id=-1`

Proponowałbym zrezygnować z `selected` i zaimplementować następujące flow:
1. Mamy zmienną `email`, którą aktualizujemy z każdą aktualizacją inputu w `Autocomplete`
2. Gdy wciskamy przycisk `Add` szukamy z listy `user'a` z odpowiednim `emailem`, a w szczególności chcemy uzyskać jego `id`
3. Wywołujemy funkcję `onProceedFunction`